### ISSUE TITLE
[BUG] HA does not start a second time

### ISSUE THREAD

**Issue by florie1706** [linuxserver/docker-homeassistant#79]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I stopped the running container which was running without a problem (using the image of yesterday, which is by now the most recent one). After I started the container again, the container did not start up healthy and in the logs I just have this message repeatedly (shown below in the log section).

### Steps To Reproduce

Do not know if this is only on my side, but an image before these env path changes (I am using this image: sha256:38320051f89a71fc3f928442e37357d998b211ccd880fe8f9fa65a419026c3f2 now) works just as expected.

### Environment

```markdown
- OS: Ubuntu 23.04
- How docker service was installed: distro's packagemanager
```

### CPU architecture

x86-64

### Docker creation

```bash
docker-compose
```

### Container logs

```bash
Traceback (most recent call last):
  File "/lsiopy/bin/hass", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/lsiopy/lib/python3.11/site-packages/homeassistant/__main__.py", line 181, in main
    args = get_arguments()
           ^^^^^^^^^^^^^^^
  File "/lsiopy/lib/python3.11/site-packages/homeassistant/__main__.py", line 81, in get_arguments
    from . import config as config_util
  File "/lsiopy/lib/python3.11/site-packages/homeassistant/config.py", line 20, in <module>
    from . import auth
  File "/lsiopy/lib/python3.11/site-packages/homeassistant/auth/__init__.py", line 4, in <module>
    import asyncio
  File "/config/lsiopy/lib/python3.11/site-packages/asyncio/__init__.py", line 21, in <module>
    from .base_events import *
  File "/config/lsiopy/lib/python3.11/site-packages/asyncio/base_events.py", line 296
    future = tasks.async(future, loop=self)
                   ^^^^^
```

**Comment by florie1706** [comment]
@Roxedus Why is it invalid?

**Comment by j0nnymoe** [comment]
No compose info - need full container log

**Comment by Hooverjc** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Current Behavior
> I stopped the running container which was running without a problem (using the image of yesterday, which is by now the most recent one). After I started the container again, the container did not start up healthy and in the logs I just have this message repeatedly (shown below in the log section).
> 
> ### Expected Behavior
> _No response_
> 
> ### Steps To Reproduce
> Do not know if this is only on my side, but an image before these env path changes (I am using this image: sha256:38320051f89a71fc3f928442e37357d998b211ccd880fe8f9fa65a419026c3f2 now) works just as expected.
> 
> ### Environment
> ```gfm
> - OS: Ubuntu 23.04
> - How docker service was installed: distro's packagemanager
> ```
> 
> ### CPU architecture
> x86-64
> 
> ### Docker creation
> ```shell
> docker-compose
> ```
> 
> ### Container logs
> ```shell
> Traceback (most recent call last):
>   File "/lsiopy/bin/hass", line 8, in <module>
>     sys.exit(main())
>              ^^^^^^
>   File "/lsiopy/lib/python3.11/site-packages/homeassistant/__main__.py", line 181, in main
>     args = get_arguments()
>            ^^^^^^^^^^^^^^^
>   File "/lsiopy/lib/python3.11/site-packages/homeassistant/__main__.py", line 81, in get_arguments
>     from . import config as config_util
>   File "/lsiopy/lib/python3.11/site-packages/homeassistant/config.py", line 20, in <module>
>     from . import auth
>   File "/lsiopy/lib/python3.11/site-packages/homeassistant/auth/__init__.py", line 4, in <module>
>     import asyncio
>   File "/config/lsiopy/lib/python3.11/site-packages/asyncio/__init__.py", line 21, in <module>
>     from .base_events import *
>   File "/config/lsiopy/lib/python3.11/site-packages/asyncio/base_events.py", line 296
>     future = tasks.async(future, loop=self)
>                    ^^^^^
> ```

same issue here, my unraid instance did a backup last night and when it restarted HA to do this, this error popped up and won't go away nor will the docker start.

[HA.txt](https://github.com/linuxserver/docker-homeassistant/files/11788690/HA.txt)

**Comment by Hooverjc** [comment]
also to add to this, I tried rolling back to each version until it worked, last version that allows HA to boot is 2023.6.2-ls156. All newer versions get stuck in the error loop above despite having worked at least once previously.

**Comment by aptalca** [comment]
You most likely installed a custom integration that lists a long deprecated package as a dependency (asyncio), a package which was integrated into python core way back in 3.4, and that old external package install breaks pip.

**Comment by aptalca** [comment]
Remove that integration and do `docker exec homeassistant bash -c 'PYTHONPATH="" pip uninstall asyncio -y'` and after that it will start properly again.

Let the integration dev know so they can remove the unnecessary dependency that breaks pip like I did here: https://github.com/jafar-atili/pyElectra/issues/1

**Comment by Hooverjc** [comment]
Thank you @aptalca for the info! 2 quick follow up questions. Would this be limited to HACS custom integrations or could it possibly include native supported integrations via HA itself? Also, any hints on how to identify the offending integration?

**Comment by aptalca** [comment]
I believe it would be a custom one. There were only two official integrations that used that as a dep, but we mitigate that by force removing it during build time.

Custom integrations however keep installing the deps on start.

But, with that said, I'm not sure if HA does a check on start if those two official integrations have their deps installed. Anyway, the official integrations that had them were `electrasmart` and `slide`. Do you have either of those enabled? If not it would be a custom integration.

**Comment by aptalca** [comment]
Those look fine, because they're not pinned to a specific version. They are set to `>=`, which means any version newer than what's listed, so it will likely get the latest unless something else has another restriction.

**Comment by florie1706** [comment]
I have a bunch of custom integrations but in non of them I found anything like a required/dependent asyncio. So I just ran your command from above to delete the deprecated package. Removed the container and let it start again. Now it works without a problem and without even touched any custom integration. I do not know where this old package came from?!?

Anyway, it works with your help. Thank you very much.
